src/sentry/static/sentry/app/components/avatarSettings.jsx

```diff
@@ -0,0 +1,515 @@
+    let file = ev.target.files[0];
+    if (!/^image\//.test(file.type)) {
+      this.setState({error: 'That is not a supported file type.'});
+      return;
+    }
+    this.state.objectURL && window.URL.revokeObjectURL(this.state.objectURL);
+    this.setState({
+      file: file,
+      objectURL: window.URL.createObjectURL(file)
+    }, () => {
+      this.props.updateDataUrlState({savedDataUrl: null});
+    });
+  },
+
+  componentWIllUnmount() {
+    this.state.objectURL && window.URL.revokeObjectURL(this.state.objectURL);
+  },
+
+  updateDimensions(ev) {
+    let $container = $(ReactDOM.findDOMNode(this.refs.cropContainer));
+    let resizeDimensions = this.state.resizeDimensions;
+    let pageY = ev.pageY;
+    let pageX = ev.pageX;
+    let top = resizeDimensions.top + (pageY - this.state.mousePosition.pageY);
+    let left = resizeDimensions.left + (pageX - this.state.mousePosition.pageX);
+    if (top < 0) {
+      top = 0;
+      pageY = this.state.mousePosition.pageY;
+    } else if (top + resizeDimensions.size > $container.height()) {
+      top = $container.height() - resizeDimensions.size;
+      pageY = this.state.mousePosition.pageY;
+    }
+    if (left < 0) {
+      left = 0;
+      pageX = this.state.mousePosition.pageX;
+    } else if (left + resizeDimensions.size > $container.width()) {
+      left = $container.width() - resizeDimensions.size;
+      pageX = this.state.mousePosition.pageX;
+    }
+    this.setState({
+      resizeDimensions: Object.assign({}, resizeDimensions, {top: top, left: left}),
+      mousePosition: {pageX: pageX, pageY: pageY}
+    });
+  },
+
+  startMove() {
+    $(document).on('mousemove', this.updateDimensions);
+    $(document).on('mouseup', this.onMouseUp);
+  },
+
+  stopMove() {
+    $(document).off('mousemove', this.updateDimensions);
+    $(document).off('mouseup', this.onMouseUp);
+    this.drawToCanvas();
+  },
+
+  onMouseDown(ev) {
+    ev.preventDefault();
+    this.setState({mousePosition: {
+      pageY: ev.pageY,
+      pageX: ev.pageX
+    }});
+    this.startMove();
+  },
+
+  onMouseUp(ev) {
+    ev.preventDefault();
+    this.stopMove();
+  },
+
+  startResize(direction, ev) {
+    ev.stopPropagation();
+    ev.preventDefault();
+    $(document).on('mousemove', this.updateSize);
+    $(document).on('mouseup', this.stopResize);
+    this.setState({
+      resizeDirection: direction,
+      mousePosition: {
+        pageY: ev.pageY,
+        pageX: ev.pageX
+      }
+    });
+  },
+
+  stopResize(ev) {
+    ev.stopPropagation();
+    ev.preventDefault();
+    $(document).off('mousemove', this.updateSize);
+    $(document).off('mouseup', this.stopResize);
+    this.drawToCanvas();
+  },
+
+  updateSize(ev) {
+    let newResizeState = {};
+    let yDiff = ev.pageY - this.state.mousePosition.pageY;
+    let xDiff = ev.pageX - this.state.mousePosition.pageX;
+    let $container = $(ReactDOM.findDOMNode(this.refs.cropContainer));
+    if (this.state.resizeDirection === 'nw') {
```

Comment:
Subjective, but I think a `switch` is more appropriate here.

```javascript
switch(this.state.resizeDirection) {
  'nw':
     ...
     break;
  'ne':
     ...
     break
  // etc
```

Or alternatively you could do:

```javascript
let method = 'getUpdatedSize' + this.state.resizeDirection.toUpperCase();
newResizeState = this[method]($container, yDiff, xDiff);
```